### DISCUSSION TITLE
What are the standards and expectations for versions and early releases within Project Reunion?

### DISCUSSION THREAD

**Discussion by mrlacey** [microsoft/WindowsAppSDK#582] [Q&A]
As Project Reunion will encompass multiple products, it will help developers working with those products if they all adopt the same standards and conventions for version numbering and how early releases are handled. This will avoid confusion and prevent developers making incorrect assumptions about those products.

Clearly documenting these (as part of the ProjectReunion docs) will also serve as a useful reference for both consumers and creators of products that are part of ProjectReunion. 

As developers we want to spend time working with the products, not trying to remember or looking up the differences.

**Version numbers**

- Will version numbers of the product be tied to the version number of ProjectReunion releases?
    - It seems this is being tried at the moment but won't work in the future when existing products with higher version numbers become part of ProjectReunion.  It will be perfectly fine if v1.x of ProjectReunion includes v4.x of a specific product.

- Does version 1.x imply a stable, supported release?
  - and v0.x still subject to change?

- How will version numbers be incremented?
  - major parts, minor parts, and beyond
  - always sequentially? if not, how and why will gaps be explained/justified?

- Will all projects follow [semver](https://semver.org/) conventions? (I know this covers above points but I wanted to call them our specifically)

**Early releases**

- Please define terms (including but not limited to) "alpha", "beta", "preview", "pre-release", "stable", "go-live", "supported", "GA", "go to market"
    - Then ensure they are used consistently.

- Please ensure that pre-release labels applied to packages use the same conventions.
    - whether this is based on words, incremental numbers, or data/time based numbering.

**Comment by mrlacey** [comment]
For clarity: I am arguing that all products under the ProjectReunion umbrella are NOT forced to have the same version number.

X-Ref: https://github.com/microsoft/microsoft-ui-xaml/issues/4508

**Comment by andrewleader** [comment]
Good question Matt, we're working on finalizing and documenting this, all of our questions are very valuable and we'll ensure that we capture all of these! We'll post an update here when we have final versions. Thank you!
